final-project/static/scripts.js

```diff
@@ -3,16 +3,29 @@
 // global variable section
 // var example = [];
 
-// execute when the DOM is fully loaded
-$(function() {
+// Enforce strict js code conventions
+"use strict";
 
-});
+// Render product on screen
+function displayProduct() {
 
-/**
- * Edit product information.
- */
-function editProduct()
-{
+  // TODO: get JSON file
+
+    // TODO: loop through JSON object to get information
+
+      // TODO: loop through array of references within the JSON object
+
+    // TODO: save JSON information to variables
+
+  // TODO: render variables on screen w/ seperate function, update()
```

Comment:
You could try a series of `.map` and `.filter`s to help you stream your data through.

Reply:
I am still trying to figure out how I am going to apply .map and .filter, but I updated to pseudocode to incorporate them a bit. They will go in the update() function instead of the displayProduct() function.

displayProduct() is now more like getProductJSON()
update() does all the work to display the JSON on screen.